src/components/ChatEntry.js

```diff
@@ -1,22 +1,40 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({ id, sender, body, timeStamp, onLikeChange }) => {
+  const [liked, setLiked] = useState(false);
+
+  const handleLike = () => {
+    const newLiked = !liked;
+    setLiked(newLiked);
+    onLikeChange(id, newLiked);
+  };
 
-const ChatEntry = (props) => {
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time">
+          <TimeStamp time={timeStamp} />
+        </p>
+        <button className="like" onClick={handleLike}>
+          {liked ? '❤️' : '🤍'}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.string.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  onLikeChange: PropTypes.func.isRequired,
```

Comment:
Because this proptype is marked `isRequired`, we're getting warnings in the tests, since the test author didn't know what you were going to call this property, and so it's absent from the props passed in during the tests. To avoid the warnings, we can leave off `isRequired`.

Ideally, we'd also include code to ensure that any "optional" values are actually present before trying to use them. For example, in the like click handler we could do something like
```js
    if (onLikeChange) {
        onLikeChange(id);  // (id is currently not being passed to this component, but should be)
    }
```
which would only try to call the function if it's not falsy (and `undefined` is falsy).

Reply:
👀 We should pass the `liked` value of the message into the ChatEntry (both in the destructuring and the prop types here). Ideally, this component shouldn't need to know how to track and update the liked status itself. Rather, it should display the status to match whatever `liked` value it receives through `props`.